network.tf

```diff
@@ -0,0 +1,35 @@
+resource "azurerm_resource_group" "rg" {
+  name     = "${var.prefix}-rg"
+  location = "West Europe"
+}
+
+resource "azurerm_virtual_network" "vnet" {
+  name                = "${var.prefix}-virtual-network"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+}
+
+resource "azurerm_subnet" "subnet" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.rg.name
+  virtual_network_name = azurerm_virtual_network.vnet.name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+locals {
+  nic_name = ["1", "2"]
+}
+
+resource "azurerm_network_interface" "nic" {
+  count               = length(local.nic_name)
```

Comment:
It's the single place where this value is used as I see. So will be better to use integer variables

Reply:
fixed

Reply:
Not fixed.

Reply:
if I understood you correct, I don't need to use local value here.
Fixed